torch/csrc/jit/passes/inliner.cpp

```diff
@@ -0,0 +1,115 @@
+#include <torch/csrc/jit/passes/inliner.h>
+#include <torch/csrc/jit/script/module.h>
+#include <torch/csrc/jit/script/error_report.h>
+
+namespace torch {
+namespace jit {
+
+namespace prim {
+using namespace ::c10::prim;
+}
+
+class Inliner {
+ public:
+  Inliner() = default;
+
+  void removeFunctionConstants(Block* b)
+  {
+    for (auto it = b->nodes().begin(); it != b->nodes().end(); it++)
+    {
+      if (it->kind() == prim::Constant && it->output()->type()->isSubclass(TypeKind::FunctionType))
+      {
+        //Inliner inlined all possible function constants
+        AT_ASSERT(it->output()->uses().size() == 0);
+        it.destroyCurrent();
+      }
+      else
+      {
+        for (auto ib : it->blocks())
+        {
+            removeFunctionConstants(ib);
+        }
+      }
+    }
+  }
+
+  void unifyIfs(Block* b)
```

Comment:
Why is this here? It is breaking loose coupling for inlining to need to think about how it would affect other statements.

Reply:
type unification is typically done by `compiler.cpp` which now doesn't have the full inlined graph (hence, no precise types) since we inline later and even later in `emitModuleHook` we compile the inlined graph which now will have more precise types, so we end up with different graphs again.